client/my-sites/patterns/pages/category/index.tsx

```diff
@@ -91,10 +91,25 @@ export const PatternsCategoryPage = ( {
 		return {
 			id: category.name || '',
 			label: category.label,
-			link: getCategoryUrlPath( category.name, patternTypeFilterFallback, false ),
+			link:
+				getCategoryUrlPath( category.name, patternTypeFilterFallback, false ) +
+				( isGridView ? '?grid=1' : '' ),
 		};
 	} );
 
+	const handleSettingView = ( value: 'grid' | 'list' ) => {
```

Comment:
Optional nit: we could move this outside the `PatternsCategoryPage` function scope, since it has no dependencies on component props or state.

Reply:
Moved outside